### ISSUE TITLE
One big schema

### ISSUE THREAD

**Issue by neopaf** [chrusty/protoc-gen-jsonschema#75]
Didn't find a way to generate not separate `.jsonschema` files, but one.

Maybe there is one?

**Comment by chrusty** [comment]
Hi @neopaf. No, there is currently no option to do this - each proto message gets its own separate JSONSchema file.

Given that each JSONSchema is a complete JSON object, writing a bunch of them to a file could make it quite difficult to use?

Could you outline the use-case you're thinking of?

**Comment by chrusty** [comment]
OK I spent some time today looking into the REF issues. Actuallly, the idea of creating a separate schema file for each message it finds is the desired behaviour.

Of course (as in your instance) you are referencing these messages within the main "party" message, but some other use cases may require these to be separate for specific validation purposes.

HOWEVER, there is something which might help you here. We actually have a CLI option to target specific messages:

```sh
protoc --jsonschema_out=messages=[Party]:. ./beeline.proto
```

I just ran this on my machine and ended up with only the one file `Party.jsonschema`
